[FYST-1946] NC: implement out of country extension question + pdf/xml

arinchoi03 opened the pull request:
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-1946
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Add a new question, populate relevant fields in NC xml/pdf
## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Unit/integration/manual tests
  - Test data used: any, just set the flipper for `extension_period` on heroku env
- After
<img width="667" alt="Screenshot 2025-04-02 at 12 16 13 PM" src="https://github.com/user-attachments/assets/fc091e30-21c0-485d-87df-42c8ba0a699a" />
<img width="589" alt="Screenshot 2025-04-02 at 12 34 13 PM" src="https://github.com/user-attachments/assets/1514a351-b647-4296-9225-ac0d6500ec2a" />
<img width="220" alt="Screenshot 2025-04-02 at 12 35 52 PM" src="https://github.com/user-attachments/assets/c9235283-d94c-45cd-9ca1-9a4cb481eeeb" />

mpidcock commented on config/locales/en.yml:
```diff
@@ -3359,6 +3359,9 @@ en:
           title: First, let's see if you can use this service to file your %{filing_year} North Carolina taxes
+      nc_out_of_country:
+        edit:
+          title: Were you out of the country on April 15, 2025 and a US citizen or resident?
```
I know the future is uncertain, but I don't want to be hardcoding filing years

arinchoi03 replied:
ooh good point

mpidcock reviewed (approved):
I think you may have added the flipper flag in more places than necessary, but I'm curious if others have thoughts so I added it to standup Qs. (not a blocker in any case)

arinchoi03 commented:
> I think you may have added the flipper flag in more places than necessary, but I'm curious if others have thoughts so I added it to standup Qs. (not a blocker in any case)

I added the flipper check in the pdf belatedly after seeing this pattern in the AZ pdf, and did briefly wonder about it. Happy to not do that in the pdf, dont feel too strongly about it either way
